src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -396,8 +396,15 @@ private async Task<CreationResultStatus> EnterTemplateManipulationFlowAsync()
             }
 
             TemplateResolutionResult templateResolutionResult = TemplateResolver.GetTemplateResolutionResult(_settingsLoader.UserTemplateCache.TemplateInfo, _hostDataLoader, _commandInput, _defaultLanguage);
-            TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _templateCreator, _hostDataLoader, _telemetryLogger, _defaultLanguage, CommandName, _inputGetter, _callbacks);
-            return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync();
+            if (templateResolutionResult.TryGetSingularInvokableMatch(out ITemplateMatchInfo templateToInvoke, out TemplateResolutionResult.Status resultStatus))
+            {
+                TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _telemetryLogger, CommandName, _inputGetter, _callbacks);
+                return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync(templateToInvoke);
```

Comment:
what would prevent this call from hanging indefinitely?
Consider specifying a CancellationToken with an inherent timeout, if there is an overload accepting a token.

Reply:
I agree here, however there is no support for cancellation anywhere down the road, so this seems out of the scope for this PR. I will create a refactoring issue for that.

Reply:
https://github.com/dotnet/templating/issues/2716